### ISSUE TITLE
[profile] Diagonal profile on large stack of image can be slow

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#864]
When drawing a profile on a large stack of images (I tested with 1k^3), the diagonal line drawing interaction becomes really slow.
It would be better if the line remains drawn interactively even if the image profile is not updated at the same rate..

**Comment by vasole** [comment]
Is there a risk of having a displayed profile not corresponding to the drawn line? Current behavior is slow but safe in the sense of what you see is what you get.

**Comment by t20100** [comment]
Yes if the slice is rendered asynchronously, we need some way to let the user know when the displayed image actually corresponds to the drawn profile line and when it is out-dated.

**Comment by payno** [comment]
This seems to came from the commit eb485717e7e5345c1941c2770494d3981065a487.
Indeed if we use PyQt4 + matplotlib on the latest version the is no slow anymore but slow down appears if using PyQt5 instead.

**Comment by t20100** [comment]
That's going to be fun...
Is it the same version of matplotlib between PyQt4 and PyQt5?

**Comment by payno** [comment]
Work with python3 + PySide2

**Comment by payno** [comment]
Hope this is fixed for latest PyQt5 version

**Comment by vallsv** [comment]
Should be tested, but with the new profile, the computation is done in a thread pool.
So it maybe mitigate this problem.
